src/main/scala/nl.knaw.dans.dd.prepub/SetVaultMetadataTask.scala

```diff
@@ -16,32 +16,81 @@
+import nl.knaw.dans.lib.error.TryExtensions
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 import nl.knaw.dans.lib.taskqueue.Task
 import org.json4s.JsonAST.JObject
 import org.json4s.jackson.{ JsonMethods, Serialization }
 
-import scala.util.Try
+import java.lang.Thread._
+import java.net.HttpURLConnection._
 import scala.util.control.NonFatal
+import scala.util.{ Failure, Success, Try }
 
-class SetVaultMetadataTask(workFlowVariables: WorkFlowVariables, dataverse: DataverseInstance, mapper: DansDataVaultMetadataBlockMapper) extends Task[WorkFlowVariables] with DebugEnhancedLogging {
+class SetVaultMetadataTask(workFlowVariables: WorkFlowVariables, dataverse: DataverseInstance, mapper: DansDataVaultMetadataBlockMapper, maxNumberOfRetries: Int, timeBetweenRetries: Int) extends Task[WorkFlowVariables] with DebugEnhancedLogging {
   private val dataset = dataverse.dataset(workFlowVariables.globalId, Option(workFlowVariables.invocationId))
 
   override def run(): Try[Unit] = {
     (for {
       _ <- dataset.awaitLock(lockType = "Workflow")
       _ <- editVaultMetadata()
-      _ <- dataverse.workflows().resume(workFlowVariables.invocationId)
-      _ = logger.info(s"Vault metadata set for dataset ${workFlowVariables.globalId}. Dataset resume called.")
+      _ <- resumeWorkflow(dataverse, workFlowVariables.invocationId, maxNumberOfRetries, timeBetweenRetries)
+      _ = logger.info(s"Vault metadata set for dataset ${ workFlowVariables.globalId }. Dataset resume called.")
     } yield ())
       .recover {
         case NonFatal(e) =>
-          logger.error(s"SetVaultMetadataTask for dataset ${workFlowVariables.globalId} failed. Resuming dataset with 'fail=true'", e)
+          logger.error(s"SetVaultMetadataTask for dataset ${ workFlowVariables.globalId } failed. Resuming dataset with 'fail=true'", e)
           dataverse.workflows().resume(workFlowVariables.invocationId, fail = true)
       }
   }
 
+  /**
+   * Executes the workflow resume call and checks the response.
+   * If the workflow has not been paused yet by dataverse a rc 404 (Not Found) is returned and after a waiting period a subsequent resume call is made.
+   * If after the maximum number of retries the workflow is still not paused, the workflow fails.
+   *
+   * @param dataverse
+   * @param invocationId
+   * @param maxNumberOfRetries
+   * @param timeBetweenRetries
+   * @return Success if the resume call succeeded, otherwise a Failure.
+   */
+  private def resumeWorkflow(dataverse: DataverseInstance, invocationId: String, maxNumberOfRetries: Int, timeBetweenRetries: Int): Try[Unit] = {
+    trace(maxNumberOfRetries, timeBetweenRetries)
+    var numberOfTimesTried = 0
+    var notPausedError = true
+
+    do {
+      val resumeResponse = dataverse.workflows().resume(invocationId)
+      notPausedError = checkResponseForPausedError(resumeResponse, invocationId).unsafeGetOrThrow
```

Comment:
I'd go for invocationIdNotFound in the function and value names. It is more literally what the error from Dataverse means and also slightly less prone to missreading (not PausedError vs notPaused Error)

Reply:
I also renamed the WorkflowNotPausedException to InvocationIdNotFoundException